src/Esprima/Utils/ToJavascriptConverter.cs

```diff
@@ -0,0 +1,1410 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using Esprima.Ast;
+
+namespace Esprima.Utils
+{
+    public class ToJavascriptConverter
+    {
+        public static string ToJavascript(Node node)
+        {
+            var visitor = new ToJavascriptConverter();
+            visitor.Visit(node);
+            return visitor.ToString();
+        }
+
+        protected StringBuilder _sb = new StringBuilder();
+        protected int _indentionLevel = 0;
+        protected bool beautify = false;
+        protected int _indentionSize = 4;
+        protected char _indentionChar = ' ';
+
+        private static readonly ConditionalWeakTable<Type, IDictionary> EnumMap = new ConditionalWeakTable<Type, IDictionary>();
+
+        private string GetEnumValue<T>(string name, T value) where T : Enum

```

Comment:
Switch case on the known operators would be sufficient, and thread-safe.

Reply:
If added Methods to convert the enum back